File: Preg3/comparar.scala
import scala.io.StdIn.readLine
import scala.collection.mutable.ListBuffer

object CompararLista {    
    def compara[A](lista1:List[A], lista2:List[A]):Boolean = {
        lista2.forall(lista1.contains)
    }  

    def main(args: Array[String]) = {
        println ("Ingrese los elementos de la primera lista, termine con una linea vacia.")
        var x = false
        val lista1 = ListBuffer[String]()
        while (!x) {
            val input = scala.io.StdIn.readLine(":").stripLineEnd
            lista1 += input
            if (input == ""){ x = true}
        }
        println ("Ingrese los elementos de la segunda lista, termine con una linea vacia.")
        x = false
        val lista2 = ListBuffer[String]()
        while (!x) {
            val input = scala.io.StdIn.readLine(":").stripLineEnd
            lista2 += input
            if (input == ""){ x = true}
        }
        if(compara(lista1.toList,lista2.toList)) {
            println ("Es subconjunto.")
        } else {
            println ("NO es subconjunto.")
        }
    }
}

File: Preg3/comparar.hs
import System.IO

main = do
    putStrLn "Ingrese la primera lista separada por espacios."
    lista1 <- getLine
    let as = map read (words lista1) :: [Int]
    putStrLn "Ingrese la segunda lista separada por cespacios."
    lista2 <- getLine
    let bs = map read (words lista2) :: [Int]
    let cs = filter (\x -> notElem x as) bs
    print bs
    if null cs then
        putStrLn "Es subconjunto de:"
    else
        putStrLn "NO es subconjunto de:"
    print as
      
